### PR TITLE
DOC: Update milestone/label policy 

### PR DESCRIPTION
**Pull request by pllim**
### Description

I came across this section and noticed that none of the words reflect how we actually use milestone and label. 😅 

Hopefully now it is more accurate.

Given now we don't render dev docs in "stable," we don't have to backport this.

### CODE REVIEW

File: docs/development/workflow/maintainer_workflow.rst [hunk 1]
```diff
@@ -142,25 +134,20 @@ followed by IPython:
...
-  * Future--this is for all issues that require attention at some point but for which no immediate solution is in
-    sight.
-
-* Bug fix release milestones should only be used for deferring issues that won't be fixed in the next minor release,
-  or for issues is previous releases that no longer apply to the mainline.
-
-* When in doubt about which milestone to use for an issue, use the next minor release--it can always be moved once
-  it's been more closely reviewed prior to release.
-
 * Active milestones associated with a specific release (eg. v0.3.0) should contain at least one issue with the
```
Comment (bsipocz) [inline docs/development/workflow/maintainer_workflow.rst:137]: I don't get this point. Github doesn't autoclose milestones (any more?). Also, I think we don't have "release label"

Comment (pllim) [reply docs/development/workflow/maintainer_workflow.rst:137]: Should I just delete this whole point? I didn't get it either but I don't understand enough about the release nitty gritty on this repo to touch it.

Comment (bsipocz) [reply docs/development/workflow/maintainer_workflow.rst:137]: 👍 on deleting, I think it is very obsolete.

Comment (pllim) [reply docs/development/workflow/maintainer_workflow.rst]: I did keep the "rolling reminder" as it is still relevant. Does it look better now?

**Review by bsipocz — APPROVED** [review]
All looks good now.

**Review by mhvk — APPROVED** [review]
Yay!  Looks consistent with our actual practice now, thanks so much!

**Comment by pllim** [comment]
Thanks for the reviews!

**Pull request merged**